usr/src/uts/common/fs/zfs/vdev_disk.c

```diff
@@ -324,10 +325,10 @@ vdev_disk_open(vdev_t *vd, uint64_t *psize, uint64_t *max_psize,
 	if (vd->vdev_devid != NULL) {
 		if (ddi_devid_str_decode(vd->vdev_devid, &dvd->vd_devid,
 		    &dvd->vd_minor) != 0) {
-			vd->vdev_stat.vs_aux = VDEV_AUX_BAD_LABEL;
 			vdev_dbgmsg(vd, "vdev_disk_open: invalid "
 			    "vdev_devid '%s'", vd->vdev_devid);
-			return (SET_ERROR(EINVAL));
+			dvd->vd_devid = NULL;
+			dvd->vd_minor = NULL;
 		}
 	}
 
```

Comment:
We've had some different issues with devids, where we wanted to bypass the devid check as well. Setting `vd->vdev_devid` to NULL also works but has the side effect of forever removing the devid from the config, which might not be ideal in some cases. I'm guessing that with this code the original devid is conserved in the config, even after successfully importing and exporting the pool?

Reply:
Right, it is preserved since it is only updated if opened by physical path and dvd->vd_devid != NULL.
It may be a good idea to update the stored devid, but this case bites me mostly when moving USB drives between Linux and illumos.